types/src/block_info.rs

```diff
@@ -63,6 +65,52 @@ impl BlockInfo {
         }
     }
 
+    /// Used to create the proposed and parent BlockInfos for the genesis QC.
+    pub fn new_for_genesis(ledger_info: &LedgerInfo) -> Self {
+        Self {
+            ledger_info.epoch(),
+            ledger_info.round(),
+            id: HashValue::zero(),
+            ledger_info.transaction_accumulator_hash(),
+            ledger_info.version(),
+            ledger_info.timestamp_usecs(),
+            next_epoch_state: None,
+        }
+    }
+
+    /// Creates a BlockInfo from a Block structure. 
+    pub fn from_block(
+        block: &Block,
```

Comment:
the entire intent was block info is public to the client but the block is not, much like block metadata so this has issues :/

Reply:
the point of this is that you HAVE to create a BlockInfo from a block, so make this change explicit instead of allowing someone to create one with new() and make a mistake